### PR TITLE
Add base icons structure

### PR DESCRIPTION
**Pull request by MMauro94**

### CODE REVIEW

File: composeApp/src/commonTest/kotlin/io/github/couchtracker/db/user/model/icon/DbIconTest.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                    }
+                }
+                context("invalid default URIs fail") {
+                    withData(
+                        "couch-tracker://icon/default",
+                        "couch-tracker://icon/default/",
+                        "couch-tracker://icon/default/%20",
+                    ) { uriString ->
+                        val uri = CouchTrackerUri(parseUri(uriString))
+                        shouldThrow<IllegalArgumentException> {
```
Comment (MMarco94) [inline composeApp/src/commonTest/kotlin/io/github/couchtracker/db/user/model/icon/DbIconTest.kt:40]: Question: is `IllegalArgumentException` the right exception for ColumnAdapter? E.g. are other implementations throwing `ParseException` or something?

File: docs/ICONS.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Icons
```
Comment (MMarco94) [inline docs/ICONS.md:1]: Maybe we should add a brief explanation on how to import a new icon.

File: composeApp/src/commonMain/kotlin/io/github/couchtracker/db/common/adapters/DbIconColumnAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package io.github.couchtracker.db.common.adapters
+
+import app.cash.sqldelight.ColumnAdapter
+import io.github.couchtracker.db.user.model.icon.DbIcon
+
+val DbIconColumnAdapter: ColumnAdapter<DbIcon, String> = CouchTrackerUriColumnAdapter.map(
```
Comment (MMarco94) [inline composeApp/src/commonMain/kotlin/io/github/couchtracker/db/common/adapters/DbIconColumnAdapter.kt:6]: Question: will this always be a couch tracker uri? Will we support custom URL or hardcoded SVGs?

File: composeApp/src/commonMain/kotlin/io/github/couchtracker/db/user/model/icon/DbIcon.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        override fun toCouchTrackerUri() = CouchTrackerUri.Authority.ICON.uri("$DEFAULT_ICON_PATH/${defaultIcon.id}")
+    }
+
+    data class UnknownDefault(val originalUri: CouchTrackerUri) : DbIcon {
+        override val icon get() = Icons.Default.BrokenImage.toIcon()
+        override fun toCouchTrackerUri() = originalUri
+    }
+
+    companion object {
+        private const val DEFAULT_ICON_PATH = "default"
```
Comment (MMarco94) [inline composeApp/src/commonMain/kotlin/io/github/couchtracker/db/user/model/icon/DbIcon.kt:29]: Question: will there ever be something other than "default"? I think it's implied, given the `couch-tracker://`

File: composeApp/src/commonTest/kotlin/io/github/couchtracker/db/user/model/icon/DbIconTest.kt (hunk 1 above)
Comment (MMauro94) [reply composeApp/src/commonTest/kotlin/io/github/couchtracker/db/user/model/icon/DbIconTest.kt:40]: I don't see any specific exception required in the documentation, I think any will do. We are already using `IllegalArgumentException` for the partial date time adapter.

For instance, the default `EnumColumnAdapter` uses `.first { }`, which throws `NoSuchElementException`: https://github.com/cashapp/sqldelight/blob/598638f0adb449ca1f3c4a30fafd20210d59c7d8/runtime/src/commonMain/kotlin/app/cash/sqldelight/EnumColumnAdapter.kt#L22

File: composeApp/src/commonMain/kotlin/io/github/couchtracker/db/common/adapters/DbIconColumnAdapter.kt (hunk 3 above)
Comment (MMauro94) [reply composeApp/src/commonMain/kotlin/io/github/couchtracker/db/common/adapters/DbIconColumnAdapter.kt:6]: The intention was to always have a `couch-tracker://` URI that can evolve as needed. For custom SVGs, for example, we could have this format: `couch-tracker://icon/svg/<base64>`

**Review by MMarco94 — APPROVED** [review]

**Pull request merged**